scapy/layers/tls/crypto/cipher_block.py

```diff
@@ -191,36 +191,40 @@ class Cipher_SEED_CBC(_BlockCipher):
 # silently not declared, and the corresponding suites will have 'usable' False.
 
 if conf.crypto_valid:
-    class _ARC2(BlockCipherAlgorithm, CipherAlgorithm):
-        name = "RC2"
-        block_size = 64
-        key_sizes = frozenset([128])
-
-        def __init__(self, key):
-            self.key = algorithms._verify_key_size(self, key)
-
-        @property
-        def key_size(self):
-            return len(self.key) * 8
-
-    _gcbn_format = "{cipher.name}-{mode.name}"
-    if GetCipherByName(_gcbn_format)(backend, _ARC2, modes.CBC) != \
-            backend._ffi.NULL:
-
+    try:
+        from cryptography.hazmat.decrepit.ciphers.algorithms import RC2
+        rc2_available = backend.cipher_supported(RC2, modes.CBC)
+    except ImportError:
+        # Legacy path for cryptography < 43.0.0
+        _gcbn_format = "{cipher.name}-{mode.name}"
+
+        class RC2(BlockCipherAlgorithm, CipherAlgorithm):
+            name = "RC2"
+            block_size = 64
+            key_sizes = frozenset([128])
+
+            def __init__(self, key):
+                self.key = algorithms._verify_key_size(self, key)
+
+            @property
+            def key_size(self):
+                return len(self.key) * 8
+        if GetCipherByName(_gcbn_format)(backend, RC2, modes.CBC) != \
+                backend._ffi.NULL:
+            rc2_available = True
+            backend.register_cipher_adapter(RC2,
+                                            modes.CBC,
+                                            GetCipherByName(_gcbn_format))
+        else:
+            rc2_available = False
+
+    if rc2_available:
         class Cipher_RC2_CBC(_BlockCipher):
-            pc_cls = _ARC2
+            pc_cls = RC2
             pc_cls_mode = modes.CBC
             block_size = 8
             key_len = 16
 
-        class Cipher_RC2_CBC_40(Cipher_RC2_CBC):
-            expanded_key_len = 16
-            key_len = 5
```

Comment:
Could you leave this? This will break `SSL_CK_RC2_128_CBC_EXPORT40_WITH_MD5`

... it appears it's not tested though.

Reply:
If you can add a test of some kind I can confirm it works with the implementation we exposed in 43 and then I’ll update the PR here. Although if it expands the key to 16 bytes itself before passing it to the RC2 constructor then it should work.

Reply:
I've added a functional test in https://github.com/secdev/scapy/pull/4286

Reply:
The classes are added to their own list within the meta class. The `_sslv2_block_cipher_algs` call you're seeing is because this class has two names depending on whether you're looking at the SSLv2 or TLS spec.

Reply:
I think we're good now 😄 Once this is merged we can migrate cipher on our side and everything should keep working for you. You'll probably want to do a release before we release 43.0.0, but we don't have a date for it (recent history suggests ~April/May though).